### ISSUE TITLE
[image_picker] pickMultiImage as a Stream

### ISSUE THREAD

**Issue by anton-definit** [flutter/flutter#116588]
Would it be possible to implement a version of `pickMultiImage` which will return a `Stream` of `XFile`?
Then we would be able to limit the number of images returned to the app - since there isn't a "maxNoOfImages" feature.

**Comment by anton-definit** [comment]
@stuartmorgan because then we will be able to cancel the stream when a certain number of images has been returned. 
Say a user selects +1000 images and we only want to allow 30 images. When using a future we will have to wait for all images to be returned before we can pick the first 30 images from the list. On my Galaxy s21 this takes several minutes.
Sorry for my rusty english 😄

**Comment by stuartmorgan** [comment]
The proposed API here would be more complicated both to implement and for plugin clients to use than just having a max number of images. Closing in favor of re-opening https://github.com/flutter/flutter/issues/85772 given that use case.
